internal/kubernetes/cluster.go

```diff
@@ -0,0 +1,54 @@
+package kubernetes
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+
+	"github.com/aws/eks-hybrid/internal/api"
+	"github.com/aws/eks-hybrid/internal/aws/eks"
+	"github.com/aws/eks-hybrid/internal/validation"
+)
+
+type ClusterProvider interface {
+	ReadClusterDetails(ctx context.Context, node *api.NodeConfig, informer validation.Informer) (*api.ClusterDetails, error)
+}
+
+type clusterProvider struct {
+	aws   aws.Config
+	cache *api.ClusterDetails
+}
+
+func NewClusterProvider(config aws.Config) ClusterProvider {
+	return &clusterProvider{
+		aws: config,
+	}
+}
+
+// ReadClusterDetails returns ClusterDetails with caching, delegating to eks.ReadClusterDetails for the actual API call
+func (p *clusterProvider) ReadClusterDetails(ctx context.Context, node *api.NodeConfig, informer validation.Informer) (*api.ClusterDetails, error) {
```

Comment:
personal preference, but i would suggest making this "validation agnostic" but keeping the informer out of here
That might lead to a bit of duplication among the different validations (calling the informer on error and what not) but I find that cleaner since it keeps all the validation framework handling in the validation itself and this just becomes a read+cache. It also allows each validation to handle an error of this method however the want.
The remediation i think it's fine to have it here since it transparent to the method signature